homeassistant/components/roborock/image.py

```diff
@@ -59,31 +65,40 @@ def __init__(
         ImageEntity.__init__(self, coordinator.hass)
         self.config_entry = config_entry
         self._attr_name = map_name
+        self._home_trait = home_trait
         self.map_flag = map_flag
-        self.cached_map = b""
+        self.cached_map: bytes | None = None
         self._attr_entity_category = EntityCategory.DIAGNOSTIC
 
-    @property
-    def is_selected(self) -> bool:
-        """Return if this map is the currently selected map."""
-        return self.map_flag == self.coordinator.current_map
-
     async def async_added_to_hass(self) -> None:
         """When entity is added to hass load any previously cached maps from disk."""
         await super().async_added_to_hass()
-        self._attr_image_last_updated = self.coordinator.maps[
-            self.map_flag
-        ].last_updated
+        self._attr_image_last_updated = self.coordinator.last_home_update
         self.async_write_ha_state()
 
+    @property
+    def _map_content(self) -> MapContent | None:
+        if self._home_trait.home_map_content and (
+            map_content := self._home_trait.home_map_content.get(self.map_flag)
+        ):
+            return map_content
+        return None
+
     def _handle_coordinator_update(self) -> None:
-        # If the coordinator has updated the map, we can update the image.
-        self._attr_image_last_updated = self.coordinator.maps[
-            self.map_flag
-        ].last_updated
+        """Handle updated data from the coordinator.
+
+        If the coordinator has updated the map, we can update the image.
+        """
+        if (map_content := self._map_content) is None:
+            return
+        if self.cached_map != map_content.image_content:
+            self.cached_map = map_content.image_content
+            self._attr_image_last_updated = self.coordinator.last_home_update
 
         super()._handle_coordinator_update()
 
     async def async_image(self) -> bytes | None:
         """Get the cached image."""
-        return self.coordinator.maps[self.map_flag].image
+        if (map_content := self._map_content) is None:
+            raise ValueError("Map flag not found in coordinator maps")
```

Comment:
I am not exactly on top of exception handling for image entities, but I would assume we should use Home Assistant Error here

Reply:
Yes, agreed, updated to use HomeAssistantError